api/src/main/java/ru/urururu/sanity/cpp/ValueParser.java

```diff
@@ -0,0 +1,11 @@
+package ru.urururu.sanity.cpp;
+
+import ru.urururu.sanity.api.CfgBuildingCtx;
+import ru.urururu.sanity.api.cfg.RValue;
+
+/**
+ * @author <a href="mailto:dmitriy.g.matveev@gmail.com">Dmitry Matveev</a>
+ */
+public abstract class ValueParser<T, V, I, B, Ctx extends CfgBuildingCtx<T, V, I, B, Ctx>> {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Convert the abstract class "ValueParser" into an interface. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1610)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00119)